src/ggmorse.cpp

```diff
@@ -1046,16 +1059,19 @@ int GGMorse::takeTxWaveformI16(WaveformI16 & dst) {
 const GGMorse::Statistics & GGMorse::getStatistics() const { return m_impl->statistics; }
 const GGMorse::Spectrogram GGMorse::getSpectrogram() const { return m_impl->stfft.spectrogram(); }
 
+#if !defined(_WIN32)
 bool GGMorse::setCharacter(const std::string & s01, char c) {
     // remove old character
-    for (auto it : m_impl->alphabet) {
-        if (it.second == c) {
-            m_impl->alphabet.erase(it.first);
-            break;
+    for (TAlphabet::iterator it = m_impl->alphabet.begin(); it != m_impl->alphabet.end(); ++it)
+    {
+        if (it->second == c) {
+            m_impl->alphabet.erase(it->first);
         }
+        break;
     }
 
     m_impl->alphabet[s01] = c;
 
     return true;
 }
+#endif
```

Comment:
This seems to change the loop logic - it will early-break on the very first iteration

Why was the `#if` necessary - does it not build on Windows?

Reply:
Yes I had this error (C2511) on MSVC for this setCharcter method. https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c2511?view=msvc-170

It seems to relate to a mismatch between the method declaration in the .h header and its definiition in the .cpp but I do not see any difference in the signature

Reply:
Secondly yes I made a mistake in the logic the break should be inside the if block.

Reply:
> Yes I had this error (C2511) on MSVC for this setCharcter method

Hm, that's odd. I think we should try to avoid the `#if`. It probably has something to do with `std::string`. Maybe a simple solution would be to change the signature to:

```c++
    bool setCharacter(const char * s01, char c);
```

Could you give this a try?

Reply:
Well spotted... It seems to work!. I just re-ran the last Windows build of SDRangel after my commit and it ran without errors.